packages/environment-check/src/checks/service-checks.ts

```diff
@@ -0,0 +1,245 @@
+    const logger = getLogger();
+    let result: ODataServiceInfo[];
+    let url: string;
+    let responseStatus: number;
+    try {
+        const catalog = provider.catalog(odataVersion);
+        result = await catalog.listServices();
+        if (result.length > 0) {
+            const numberOfServices = countNumberOfServices(result);
+            logger.info(
+                t('info.numServicesForSystem', {
+                    odataVersion,
+                    system: systemName,
+                    numServicesForSystem: getServiceCountText(numberOfServices)
+                })
+            );
+        }
+    } catch (error) {
+        responseStatus = error?.response?.status || error?.cause?.status;
+        logger.error(
+            catalogMessages[responseStatus]
+                ? catalogMessages[responseStatus](systemName, odataVersion)
+                : t('error.queryFailure', { odataVersion, system: systemName })
+        );
+        const errorJson = error.toJSON ? error.toJSON() : {};
+        if (errorJson?.config?.auth?.password) {
+            delete errorJson.config.auth.password;
+        }
+        logger.debug(t('error.urlRequestFailure', { url, error: error.message, errorObj: error }));
+    }
+    return {
+        messages: logger.getMessages(),
+        result,
+        responseStatus
+    };
+}
+
+/**
+ * Checks if ATO catalog is available.
+ *
+ * @param provider the AbapServiceProvider to be used
+ * @returns Result messages and result of ato catalog request
+ */
+export async function checkAtoCatalog(
+    provider: AbapServiceProvider
+): Promise<{ messages: ResultMessage[]; isAtoCatalog: boolean }> {
+    const logger = getLogger();
+    let isAtoCatalog = false;
+    try {
+        const atoService = await provider.getAdtService<AtoService>(AtoService);
+        if (Object.keys(atoService).length) {
+            isAtoCatalog = true;
+            logger.info(t('info.atoCatalogAvailable'));
+        } else {
+            logger.warn(t('warning.atoCatalogNotAvailable'));
+        }
+    } catch (e) {
+        logger.error(t('error.atoCatalogError'));
+        logger.debug(e.message);
+    }
+    return {
+        messages: logger.getMessages(),
+        isAtoCatalog
+    };
+}
+
+/**
+ * Checks for an existing instance of the UI5 ABAP repository service or creates one.
+ *
+ * @param provider the AbapServiceProvider to be used
+ * @returns Result messages and result of the request
+ */
+export async function checkUi5AbapRepository(
+    provider: AbapServiceProvider
+): Promise<{ messages: ResultMessage[]; isSapUi5Repo: boolean }> {
+    const logger = getLogger();
+    let isSapUi5Repo = false;
+    try {
+        const sapUI5Repo = provider.getUi5AbapRepository();
+        const response = await sapUI5Repo.get('', {
+            headers: {
+                Accept: 'application/*'
+            }
+        });
+        if (response.status !== 404) {
```

Comment:
It might be easier to just log the response status and response message (if any). This if condition hides the original message and might not provide accurate diagnostics.

Reply:
For the markdown we send a boolean value whether the check was succesful or not, this determines whether the check is presented with a green or red indicator. 
I've updated to be set to **true** for a 200 response, but **false** for anything else.
 But, the message will say the check could not be determined and to check the logs (rather than unavailable). 
Regardless, the status and message will be logged in both cases as suggested.

Reply:
I think is a much improved approach, thanks for the suggestion @slin-sap